translib/transformer/subscribe_req_xlate.go

```diff
@@ -0,0 +1,2135 @@
+
+	return nil
+}
+
+func (reqXlator *subscribeReqXlator) Translate(targetOnly bool) error {
+
+	if reqXlator == nil {
+		log.Warningf("Translate: subscribeReqXlator is nil")
+		return tlerr.InternalError{Format: "subscribeReqXlator is nil"}
+	}
+
+	if log.V(dbLgLvl) {
+		log.Infof("%v Translate: reqXlator: ", reqXlator.subReq.reqLogId, *reqXlator.subReq)
+	}
+
+	var err error
+	ygXpathInfoTrgt, ok := xYangSpecMap[reqXlator.subReq.ygPath]
+	if !ok || ygXpathInfoTrgt == nil {
+		log.Warningf("%v Translate: ygXpathInfo data not found in the xYangSpecMap for xpath : %v", reqXlator.subReq.reqLogId, reqXlator.subReq.ygPath)
+		return tlerr.InternalError{Format: "Error in processing the subscribe path", Path: reqXlator.subReq.reqUri}
+	} else if _, ygErr := getYgEntry(reqXlator.subReq.reqLogId, ygXpathInfoTrgt, reqXlator.subReq.ygPath); ygErr != nil {
+		return ygErr
+	} else {
+		reqXlator.subReq.subReqXlateInfo.ygXpathTrgtInfo = ygXpathInfoTrgt
+		if err = reqXlator.processSubscribePath(); err != nil {
+			log.Warningf("%v Translate: Error in processSubscribePath for the path: %v, and the error is %v", reqXlator.subReq.reqLogId, reqXlator.subReq.reqUri, err)
+			return tlerr.NotSupportedError{Format: "Error in processing the subscribe path", Path: reqXlator.subReq.reqUri}
+		}
+	}
+
+	if err = reqXlator.translateTargetNodePath(ygXpathInfoTrgt); err != nil {
+		log.Warning(reqXlator.subReq.reqLogId, "Translate: Error in translating the target node subscribe path: ", err)
+		return err
+	}
+	if !targetOnly {
+		if err = reqXlator.translateChildNodePaths(ygXpathInfoTrgt); err != nil {
+			log.Warning(reqXlator.subReq.reqLogId, "Translate: Error in translating the child node for the subscribe path: ", err)
+			return err
+		}
+	}
+
+	trgtPathXlateInfo := reqXlator.subReq.subReqXlateInfo.TrgtPathInfo
+	if trgtPathXlateInfo.PType == Sample && trgtPathXlateInfo.hasDbTableInfo() {
+		if reqXlator.trgtYgNode != nil && trgtPathXlateInfo.MinInterval < reqXlator.trgtYgNode.subMinIntrvl {
+			trgtPathXlateInfo.MinInterval = reqXlator.trgtYgNode.subMinIntrvl
+		} else if trgtPathXlateInfo.MinInterval == 0 {
+			trgtPathXlateInfo.MinInterval = apis.SAMPLE_NOTIFICATION_MIN_INTERVAL // default value; in seconds
+		}
+		if log.V(dbLgLvl) {
+			log.Infof("%v Translate: requested target path xlate info: Min Interval: %v for the path: %v", reqXlator.subReq.reqLogId, trgtPathXlateInfo.MinInterval, reqXlator.subReq.reqUri)
+		}
+	}
+
+	return err
+}
+
+func (reqXlator *subscribeReqXlator) translateTargetNodePath(trgtYgxPath *yangXpathInfo) error {
+	if trgtPathXlator, err := reqXlator.getSubscribePathXlator(reqXlator.subReq.gPath, reqXlator.subReq.reqUri,
+		trgtYgxPath, nil, &ygXpathNode{ygXpathInfo: trgtYgxPath, ygPath: reqXlator.subReq.ygPath,
+			dbFldYgPathMap: make(map[string]string), dbTblFldYgPathMap: make(map[string]map[string]string)}); err != nil {
+		log.Warning(reqXlator.subReq.reqLogId, "Error in getSubscribePathXlator: error: ", err)
+		return err
+	} else {
```

Comment:
Consider removing this else block since code returns in the above if block http://google.github.io/styleguide/go/decisions#indent-error-flow